source/uwp/Renderer/lib/AdaptiveImageRenderer.cpp

```diff
@@ -666,4 +738,93 @@ namespace AdaptiveCards::Rendering::Uwp
+            bool isHeightSet = false;
+
+            if (!height.empty())
+            {
+                if (auto heightAsDouble = TryHStringToDouble(height))
+                {
+                    SetRasterizedPixelHeightAsync(imageSource, heightAsDouble.value());
+                    isHeightSet = true;
+                }
+            }
+
+            if (!width.empty())
+            {
+                if (auto widthAsDouble = TryHStringToDouble(width))
+                {
+                    SetRasterizedPixelWidthAsync(imageSource, widthAsDouble.value(), isHeightSet);
+                }
+            }
+        }
+    }
+
+    winrt::fire_and_forget render_xaml::XamlBuilder::SetRasterizedPixelHeightAsync(winrt::SvgImageSource const imageSource,
+                                                                           double const imageSize,
+                                                                           bool const dropIfUnset)
+    {
+        co_await winrt::resume_foreground(imageSource.Dispatcher());
+        auto currentSize = imageSource.RasterizePixelHeight();
+        bool sizeIsUnset = isinf(currentSize);
+
+        bool dropHeight = sizeIsUnset && dropIfUnset;
+
+        if (!dropHeight)
+        {
+            imageSource.RasterizePixelHeight(imageSize);
+        }
+    }
+
+    winrt::fire_and_forget render_xaml::XamlBuilder::SetRasterizedPixelWidthAsync(winrt::SvgImageSource const imageSource,
+                                                                          double const imageSize,
+                                                                          bool const dropIfUnset)
+    {
+        co_await winrt::resume_foreground(imageSource.Dispatcher());
+        auto currentSize = imageSource.RasterizePixelWidth();
+        bool sizeIsUnset = isinf(currentSize);
+
+        bool dropWidth = sizeIsUnset && dropIfUnset;
+
+        if (!dropWidth)
+        {
+            imageSource.RasterizePixelWidth(imageSize);
+        }
+    }
+
+    void render_xaml::XamlBuilder::SetRasterizedPixelHeight(winrt::ImageSource const& imageSource, double const& imageSize) {
+        if (auto image = imageSource.try_as<winrt::SvgImageSource>())
+        {
+            auto currentSize = image.RasterizePixelHeight();
+            bool sizeIsUnset = isinf(currentSize);
+
+            if (sizeIsUnset)
+            {
+                image.RasterizePixelHeight(imageSize);
+            }
+        }
+    }
+
+    void render_xaml::XamlBuilder::SetRasterizedPixelWidth(winrt::ImageSource const& imageSource, double const& imageSize) {
+        if (auto image = imageSource.try_as<winrt::SvgImageSource>())
+        {
+            auto currentSize = image.RasterizePixelWidth();
+            bool sizeIsUnset = isinf(currentSize);
+
+            if (sizeIsUnset)
+            {
+                image.RasterizePixelWidth(imageSize);
+            }
+        }
+    }
```

Comment:
don't duplicate the method just to overload the parameter :)
simply add `dropIfUnset` with default value

Reply:
These two methods are only called from `BuildImage` and do not require switching to the foreground thread. So we can remove that logic and keep them sync.

I also include an imagesource check here, since it isn't confirmed that they are svgs at this point.